src/tests/Common/Directory.Build.targets

```
@@ -36,6 +37,14 @@
   <Target Name="CopyDependencyToCoreRoot"
           DependsOnTargets="ResolveAssemblyReferences;ResolveRuntimeFilesFromLocalBuild">
 
+    <!-- Publish crossgen2 on supported platforms. -->
+
+    <MSBuild Condition="'$(Crossgen2Supported)' == 'true'"
+      Targets="Restore;Publish"
+      BuildInParallel="true"
+      Properties="PublishDir=$(CORE_ROOT)\crossgen2;DisableSourceLink=true"
+      Projects="$(InstallerProjectRoot)pkg\sfx\Microsoft.NETCore.App\Microsoft.NETCore.App.Crossgen2.sfxproj" />
```

If you use the PublishToDisk target, you'll get the same layout as the tarball/zip has. Additionally, if we ever make the managed test build act as a subset in the root build scripts (instead of being a separate script), the build will correctly dedupe against a build of the sfxproj directly.

```suggestion
    <MSBuild Condition="'$(Crossgen2Supported)' == 'true'"
      Targets="PublishToDisk"
      BuildInParallel="true"
      Properties="OutputPath=$(CORE_ROOT)\crossgen2"
      Projects="$(InstallerProjectRoot)pkg\sfx\Microsoft.NETCore.App\Microsoft.NETCore.App.Crossgen2.sfxproj" />
```

---

Looks like we need to pass additional props for Publish->PublishToDisk, becasue tests are not resolving NativeAotSupported anymore. The preference in crossgen2 publishing used to be in this order: PublishAot->PublishReadyToRun->PublishSingleFile based on the target platform. That is apparently not working after the rebase (and all platforms are looking for apphost).